R/derive_summary_records.R

```diff
@@ -196,12 +196,39 @@ derive_summary_records <- function(dataset,
     subset_ds <- dataset
   }
 
+  # Summarise the analysis value and bind to the original dataset
   # Summarise the analysis value and bind to the original dataset
   bind_rows(
     dataset,
-    subset_ds %>%
-      group_by(!!!by_vars) %>%
-      summarise(!!analysis_var := summary_fun(!!analysis_var)) %>%
-      mutate(!!!set_values_to)
+    get_summary_records(
+      dataset,
+      by_vars = by_vars,
+      filter = !!filter,
+      analysis_var = !!analysis_var,
+      summary_fun = summary_fun,
+      set_values_to = set_values_to)
   )
+  get_summary_records <- function(dataset,
+                                  by_vars,
+                                  filter,
+                                  analysis_var,
+                                  summary_fun,
+                                  set_values_to){
+  # Apply filter
+  if (!quo_is_null(filter)) {
+    subset_ds <- dataset %>%
+      group_by(!!!by_vars) %>%
+      filter(!!filter) %>%
+      ungroup()
+  } else {
+    subset_ds <- dataset
+  }
+
+  # Summarise the analysis value
+  subset_ds %>%
+    group_by(!!!by_vars) %>%
+    summarise(!!analysis_var := summary_fun(!!analysis_var)) %>%
+    mutate(!!!set_values_to)
+
```

Comment:
Please use the simplified version I proposed in my last change request.

Reply:
Updated the code please check.

Reply:
Please use
```
  # Summarise the analysis value
  dataset %>%
    group_by(!!!by_vars) %>%
    filter_if(!!filter) %>%
    summarise(!!analysis_var := summary_fun(!!analysis_var)) %>%
    mutate(!!!set_values_to) %>%
    ungroup()
```